src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java

```diff
@@ -11,42 +11,108 @@
 import com.wikia.webdriver.elements.common.Navigate;
 import com.wikia.webdriver.elements.mercury.components.Search;
 import com.wikia.webdriver.elements.mercury.components.TopBar;
+import com.wikia.webdriver.elements.mercury.pages.SearchResultsPage;
 
+import lombok.Getter;
+import org.openqa.selenium.WebDriverException;
 import org.testng.annotations.Test;
 
 @Execute(onWikia = MercuryWikis.MERCURY_AUTOMATION_TESTING)
 @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+@Test(groups = "MercurySearch")
 public class SearchTests extends NewTestTemplate {
 
   private TopBar topBar;
-  private Search search;
+
+  @Getter(lazy = true)
+  private final Search search = new Search();
+
+  private static final String SEARCH_PHRASE = "Infobox";
+  private static final String SEARCH_PHRASE_NO_RESULTS = "DubiousQueryWithNoResults";
 
   private void init() {
     this.topBar = new TopBar(driver);
-    this.search = new Search();
-
     new Navigate().toPage(MercurySubpages.MAIN_PAGE);
   }
 
-  @Test(groups = "mercury_search_navigateToPageUsingSearch")
-  public void mercury_search_navigateToPageUsingSearch() {
-    String searchingPhrase = "Infobox";
+  @Test(groups = "mercury_search_navigateToPageUsingSearchSuggestions")
+  public void mercury_search_navigateToPageUsingSearchSuggestions() {
+    init();
+
+    try {
+      this.topBar.openSearch().typeInSearch(SEARCH_PHRASE).selectSearchSuggestion(0);
+    } catch (Exception e) {
+      System.out.println(e);
+    }
+
+    Assertion.assertTrue(driver.getCurrentUrl().contains(SEARCH_PHRASE));
+  }
+
+  @Test(groups = "mercury_search_navigateToPageUsingSearchResults")
+  public void mercury_search_navigateToPageUsingSearchResults() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)